src/object/cli_obj.c

```diff
@@ -439,6 +470,17 @@ obj_dkeyhash2shard(struct dc_object *obj, uint64_t hash, unsigned int map_ver,
 	grp_size = obj_get_grp_size(obj);
 	idx = hash % grp_size + grp_idx * grp_size;
 
+	if (!to_leader && obj->cob_time_fetch_leader != NULL &&
+	    obj->cob_time_fetch_leader[grp_idx] != 0) {
+		int		rc;
+		struct timespec	now;
+
+		rc = clock_gettime(CLOCK_MONOTONIC_COARSE, &now);
+		if (rc == 0 && OBJ_FETCH_LEADER_INTERVAL >=
+		    now.tv_sec - obj->cob_time_fetch_leader[grp_idx])
```

Comment:
probably can reuse gurt's d_gettime() + d_time2s()

Reply:
#define _gurt_gettime(ts) clock_gettime(CLOCK_MONOTONIC, ts)

We do not need very precise time here, the efficiency is more important since it is on the IO critical path. The CLOCK_MONOTONIC_COARSE seems more faster.